Catappult/appcoins-wallet-android
# Feature/eskl 302 onboarding + 301 referral

**What does this PR do?**

This PR adds 2 screens to eskills flow: 
-  Onboarding screen for new users, with a place to redeem a referral code 
-  Improvement on 'Waiting for opponents' screen where user's that have already played a match (LIVE env) have access to their referral code to share.

**Database changed?**

No

**Where should the reviewer start?**
SkillsFragment
**How should this be manually tested?**
-> New device and new wallet, test onboarding functionality 
    1. Try inserting a random referral code  (should see correct error message)
    2. Insert **472323C0**, should jump to next screen, 1$ should be sent to your wallet by our provider, during your game.
    3. Coming back to this screen after a game, the onboarding should not appear, and the normal buy screen should pop up.
-> With a Topped up wallet and device with at least a game:
   1. Buy a ticket
   2. You should see in the waiting for opponents screen your referral to share, make sure the functionalities work (such as Share and tooltip) 
   

**What are the relevant tickets?**

  Tickets related to this pull-request: [ESKL-294](https://aptoide.atlassian.net/browse/ESKL-294)

**Code Review Checklist**
